src/hex_http.erl

```diff
@@ -0,0 +1,27 @@
+-module(hex_http).
+-export([get/3]).
+
+-type adapter() :: module().
+-type status() :: non_neg_integer().
+-type headers() :: map().
+
+-callback get(URI :: string(), headers()) ->
+    {ok, status(), headers(), binary()} |
+    {error, term()}.
+-callback user_agent_string() -> string().
+
+-spec get(adapter(), string(), headers()) -> {ok, {status(), headers(), binary()}} | {error, term()}.
+get(Adapter, URI, Headers) ->
+    Headers2 = put_new("user-agent", user_agent(Adapter), Headers),
+    Adapter:get(URI, Headers2).
+
+user_agent(Adapter) ->
+    OtpVersion = erlang:system_info(otp_release),
```

Comment:
Can we get a more specific OTP version? You can check how hex or rebar3 get this.